objects/src/accounts/storage.rs

```diff
@@ -71,11 +71,21 @@ impl AccountStorage {
         &self.slots
     }
 
-    /// Returns a reference to the Merkle store that backs the storage.
+    /// Returns a mutable reference to the sparse Merkle tree that backs the storage slots.
+    pub fn slots_mut(&mut self) -> &mut SimpleSmt {
+        &mut self.slots
+    }
```

Comment:
I'd like to move away from exposing the underlying details of the implementation publicly. Since we already do this via `slots()` and `store()` methods, it's probably fine to add their `*_mut` counterparts. But also, if there is a way to avoid it (e.g., rely on `set_item()`) that would probably be better.

Also, another note. I probably wouldn't worry about mocking out the `store` part as it will likely get refactored soon.